src/main/java/cz/xtf/openshift/OpenShiftUtil.java

```diff
@@ -828,4 +773,97 @@ private Path storeLog(String log, Path dirPath, String fileName) throws IOExcept
 
 		return filePath;
 	}
+
+	private OpenShiftWaiters waiters = new OpenShiftWaiters();
+
+	// Waiting
+	public OpenShiftWaiters waiters() {
+		return waiters;
+	}
+
+	public class OpenShiftWaiters {
+
+		private OpenShiftWaiters() {
+
+		}
+
+		public void buildCompletion(String buildName) throws TimeoutException, InterruptedException {
+			log.info("Waiting for completion of build {}", buildName);
+			boolean success = Waiters.waitFor(() -> getBuild(buildName).getStatus().getPhase(), "Complete"::equals, "Failed"::equals, 600_000L, 5_000L);
+
+			if(!success) throw new IllegalStateException("Build " + buildName + " has failed");
+		}
+
+		public SimpleWaiter buildFinished(String buildName) {
+			BooleanSupplier bs = () -> getBuild(buildName).getStatus().getPhase().matches("Complete|Failed");
+			return Waiters.waitFor(bs, 600_000).reason("Waiting till build " + buildName + " finishes", Waiters.LogPoint.BOTH).interval(5_000L);
+		}
+
+		public SimpleWaiter cleanProject() {
+			BooleanSupplier bs = () -> {
+				boolean noDc = getDeploymentConfigs().isEmpty();
+				boolean noRc = client.replicationControllers().list().getItems().isEmpty();
+				boolean noBc = getBuildConfigs().isEmpty();
+				boolean noIs = getImageStreams().isEmpty();
+				boolean noEp = getEndpoints().isEmpty();
+				boolean noSvc = getServices().isEmpty();
+				boolean noBuild = getBuilds().isEmpty();
+				boolean noRoute = getRoutes().isEmpty();
+				boolean noPod = getPods().isEmpty();
+				boolean noPvc = getPersistentVolumeClaims().isEmpty();
+				boolean noHpa = getHorizontalPodAutoscalers().isEmpty();
+				boolean noCm = getConfigMaps().isEmpty();
+				boolean noUserSecrets = getUserSecrets().isEmpty();
+				boolean noUserSa = getUserServiceAccounts().isEmpty();
+
+				return noDc && noRc && noBc && noIs && noEp && noSvc && noBuild && noRoute && noPod && noPvc && noHpa && noCm && noUserSecrets && noUserSa;
```

Comment:
Maybe put those booleans into array and evaluate at once? :)

Reply:
Good point, updated.